### DISCUSSION TITLE
Resource Update via Webhook request only

### DISCUSSION THREAD

**Discussion by nickperkins** [fluxcd/flux2#2060] [Q&A] [answered]
We are using image automation to deploy new dev and staging versions of our cluster deployments. For our production environment, we would like to limit these to only occur when we trigger a webhook to request reconciliation. Is this configuration possible?

**Comment by kingdonb** [comment, marked as answer]
No, it's not currently possible, and I don't think it would be possible as the design of reconcilers generally hinges around the expectation that they are all going to be reconciled against some interval, unless they are suspended.

This is in keeping with the GitOps Principles that have been developed and published at https://opengitops.dev by the GitOps Working Group. In order for reconciliation to be considered as continuous, it should be on some interval that cannot be disabled.

Have you considered using the `imageUpdateAutomation.spec.git.push.branch` to push your changes to a different branch, then perhaps something like [this workflow](https://fluxcd.io/docs/use-cases/gh-actions-auto-pr/) to raise them to the attention of someone who can approve and merge them into production? This is a more git-centric workflow that hopefully does everything you're looking for, but without the addition of a webhook requirement or any changes to the protocols of Flux itself.

(If that doesn't work for you, I'm interested to know why!)

**Reply by kingdonb** [reply]
The other alternative that I may have overlooked, is to target a specific `ImageUpdateAutomation` to production-only manifest paths, and suspend that IUA whenever it should not be updating. Then only resume and suspend again when the needed updates have been completed. This still does not seem like a good solution to me, but it is closer to what was asked for than the solution that I proposed.

**Reply by nickperkins** [reply]
Thanks @kingdonb. This is what we are looking to do. I was hoping to trigger the webhook via a gitlab CI pipeline but instead I'm now creating a MR.
